op-batcher/batcher/channel_manager_test.go

```diff
@@ -106,24 +108,193 @@ func TestChannelManager_TxResend(t *testing.T) {
 
 	txdata0, err := m.TxData(eth.BlockID{})
 	require.NoError(err)
-	txdata0bytes := txdata0.Bytes()
-	data0 := make([]byte, len(txdata0bytes))
+
+	// confirm one frame to keep the channel open
+	m.TxConfirmed(txdata0.ID(), eth.BlockID{})
+
+	txdata1, err := m.TxData(eth.BlockID{})
+	require.NoError(err)
+	txdata1bytes := txdata1.Bytes()
+	data1 := make([]byte, len(txdata1bytes))
```

Comment:
This change was necessary because I altered the behavior of `TxFailed` to reset the channel entirely if the channel had no pending or confirmed transactions. A channel reset would result in a new channel ID, and thus different transaction data. I can probably revert these test changes if we only clear the channel in `TxFailed` if the `ChannelManager` is already closed.

Reply:
I think that makes sense, only clearing if closed.